frontend/src/components/Page.jsx

```diff
@@ -1,71 +1,100 @@
-import React, { useContext, useState } from "react";
+import React, { useContext } from "react";
 
 import Navbar from "./Navbar";
 import userContext from "../contexts/context";
 import Form from "./Form";
-import DishIndex from "./Dish/DishIndex";
+import DishItem from "./Dish/DishItem";
+import BranchItem from "./Branch/BranchItem";
+import BusinessItem from "./Business/BusinessItem";
 
+const _ = require("lodash");
 const fields = require("../fields/index");
-const apiUtil = require("../util/apiUtil");
+const validator = require("../util/validations");
 
-function Page() {
-  const { currentUrl, setUrl, toggleModal, isModalOpen } = useContext(
+function Page({
+  formInfo,
+  sendForm,
+  isAdmin,
+  prepareForm,
+  currentUrl,
+  entitiesState,
+  setEntities,
+  setCurrentUrl,
+}) {
+
+  const { toggleModal, isModalOpen, language } = useContext(
     userContext
   );
-  const [formInfo, setFormInfo] = useState({
-    entityName: "dish",
-    entity: "",
-    postOrPut: "post",
-  });
 
   function dish() {
-    debugger;
-    setUrl("/dish");
+    setCurrentUrl("dish")
+  }
+
+  function branch() {
+    setCurrentUrl("branch")
   }
 
-  function sendForm(formDetails) {
-    debugger
-    if (formInfo.postOrPut === "post") {
-      apiUtil[formInfo.postOrPut](
-        `/${formInfo.entityName}`,
-        formDetails,
-        (data) => {
-          debugger
-          console.log(data);
-        },
-        (err) => {
-          debugger
-          console.log(err);
-        }
-      );
-    } else {
-      apiUtil[formInfo.postOrPut](
-        `/${formInfo.entityName}/${formInfo.entity._id}`,
-        formDetails,
-        (data) => {
-          debugger
-          console.log(data);
-        },
-        (err) => {
-          debugger
-          console.log(err);
-        }
-      );
-    }
+  function business() {
+    setCurrentUrl("business")
   }
 
   return (
     <div>
       <Navbar />
       <button onClick={dish}>Click dish</button>
-      {currentUrl === "/dish" && <DishIndex setFormInfo={setFormInfo} />}
+      <button onClick={branch}>Click branch</button>
+      <button onClick={business}>Click business</button>
+      {isAdmin && !validator.foreignKeysValidator(currentUrl) ? (
+        <div>
+          <button
+            onClick={() => prepareForm("post", {}, currentUrl, {})}
+          >{`Create ${_.capitalize(currentUrl)}`}</button>
+        </div>
+      ) : (
+        ""
+      )}
+      {currentUrl === "dish" &&
```

Comment:
Since this render function will grow bigger very fast, I think let's extract each of these to its own component (the code after each `&&`. (`Dishes.jsx`, `Branches.jsx`, etc)